src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/DoNotInitializeUnnecessarily.cs

```diff
@@ -46,7 +46,7 @@ public sealed override void Initialize(AnalysisContext context)
                     !field.IsConst &&
                     init.Value != null &&
                     field.GetAttributes().IsEmpty && // in case of attributes that impact nullability analysis
-                    UsesKnownDefaultValue(init.Value, field.Type))
+                    UsesKnownDefaultValue(init.Value, field.Type, field.ContainingType))
```

Comment:
Why is this focusing on the field's containing type? This doesn't look right. In C# 10, any struct can define a parameterless ctor, the issue isn't just when the field is of a type it's a part of.

Reply:
📝 See https://github.com/DotNetAnalyzers/StyleCopAnalyzers/pull/3432 for how a similar situation was handled, in case we want to do the same here.

Reply:
@stephentoub I didn't quite understand when the change was actually meaningful (@sharwell comment on StyleCop ticket helped me).